### DISCUSSION TITLE
Luba w/o Internet

### DISCUSSION THREAD

**Discussion by MeMichael151** [mikey0000/PyMammotion#11] [General]
Hi, with great interest I follow this project for the HA integration. I have no dev skills myself, but if I can help, let me know. 

I'm looking to get the Luba2 with the camera system. Because of that, I'd like to know, what kind of data is exchanged with the Mammotion Servers. 
And whether the Luba cares if I block its Internet access. 

I have WiFi on my property, so it shall use that to communicate with HA. But no Internet. 

Lastly, can FW updates be downloaded manually from the servers and then be applied locally?

**Comment by mikey0000** [comment]
So as of right now there is no local wifi access, only  Bluetooth. FW updates are directly downloaded to Luba from what I can tell, I've watched the network traffic and read the reverse engineered code for FW updates, nothing goes through the phone. I've almost sussed WiFi access from pyluba but again, not local. Creating local control is no easy feat over wifi

**Comment by mikey0000** [comment]
My recommendation is to create Bluetooth proxies with Esp32's I should get onto at least creating the Bluetooth integration for a start

**Reply by MeMichael151** [reply]
What do these Bluetooth Proxy ESP32s do? Is HA connecting to the ESPs via WiFi and the ESPs then control the Luba via Bluetooth?

Would you put one on the Luba that then connect to the WiFi or place many on the property?

**Reply by MeMichael151** [reply]
I reached out to Luba requesting a local API as well. Will keep you posted

**Reply by MeMichael151** [reply]
Oh and one other question: What is communicated to the Luba servers? FW Update, Location and Usage Statistics or also Camera/Microphone feeds? 😬
And where are those Mammotion Servers located. 
My router does DPI and I filtered out some countries... just wondering if I might run into issues with that one I have my Luba

**Reply by mikey0000** [reply]
Mammotion Servers are  in AWS so could be located in multiple regions.

**Comment by mikey0000** [comment]
Luba has an esp32 inside it, it has Bluetooth Le. Bluetooth proxies proxy Bluetooth signals back to HA and vice versa. Api servers are regional so depending on your region will decide where the server is located. Over WiFi all traffic is sent. They do not have local wifi API access only Bluetooth. 

The proxies are to extend Bluetooth coverage.

As for the camera I only have a Luba 1, so I'd need to investigate the app. But mammotion have their own servers as well as using alibabas for device iot functions

**Comment by vertgo** [comment]
Wait. Can I use this Bluetooth proxy to drive the robot when I am not local?

**Reply by jLynx** [reply]
Yes, but not currently
